Support default docker and weak docker profiles

indradhanush opened the pull request:
The weak docker profile is not complete, but with this PR the user can now select from a dropdown of available docker profiles, which lets the container be run with non-root or root user.

indradhanush committed:
docker: Add support for docker profiles

To start with we have default and weak docker profiles where the user
is non-root and root respectively.

surajssd reviewed (approved):
LGTM

alban commented on docker.go:
```diff
@@ -76,6 +76,7 @@ func (h *handler) startContainer(ctrInfo containerInfo) (string, *websocket.Conn
 	r, err := h.dcli.ContainerCreate(
 		context.Background(),
 		&container.Config{
+			User:         "nobody",
```
That's good for the default container profile, but I wonder about the userns profile:
- if we use userns, we're already shifting the uids, so using root should be fine, since it's not the real root
- how is the string "nobody" converted to an uid? Without userns, I would think it uses `/etc/passwd` of the host, but that mapping should not be relevant in a different userns.

Also, on my Fedora laptop, nobody is uid 99:
```
$ grep nobody /etc/passwd
nobody:x:99:99:Nobody:/:/sbin/nologin
nfsnobody:x:65534:65534:Anonymous NFS User:/var/lib/nfs:/sbin/nologin
```

I would not want nobody to be mapped to 65534 because of the special meaning it has with in userns (see `man user_namespaces`) but in some distro, nobody could be mapped to 65534...

indradhanush committed:
frontend: Add dropdown to select profile

indradhanush committed:
server: Handle profile input from frontend

We now populate the containerInfo struct with the appropriate value
for the dockerProfile attribute.

alban commented on frontend/index.html:
```diff
@@ -31,6 +31,12 @@ <h1>Profiles</h1>
+            <select name="profile">
+              <option value="default-docker">default-docker</option>
+              <option value="weak-docker">weak-docker</option>
+            </select>
```
Maybe a link to the profiles documentation:
https://github.com/kinvolk/container-escape-bounty/blob/master/Documentation/profiles.md

alban commented on server.go:
```diff
@@ -58,9 +58,15 @@ func constructContainerInfo(r *http.Request) containerInfo {
 	}
+
+	if len(r.URL.Query()["profile"]) > 0 {
+		c.dockerProfile = dockerProfile(r.URL.Query()["profile"][0])
```
Could we do some validation of the "profile" parameter, if the user gives something else than `defaultDockerProfile` or `weakDockerProfile`? At the moment, it does not seem problematic, but it will be more difficult to assess when we add more parameters & profiles.

alban commented on frontend/index.html:
```diff
@@ -31,6 +31,15 @@ <h1>Profiles</h1>
+            Profile:
+            <!-- Profile is an abstraction to support different configuration sets for running -->
+            <!-- containers. More information is available here about the supported profiles and their meanings: -->
+            <!-- https://github.com/kinvolk/container-escape-bounty/blob/master/Documentation/profiles.md -->
```
Oh, I meant something visible for the security researcher, something like:
```
<a href="https://github.com/kinvolk/container-escape-bounty/blob/master/Documentation/profiles.md">
  Profile:
</a>
```

alban reviewed (approved):
1 more comment, LGTM after that
